geniza/corpus/management/commands/add_links.py

```diff
@@ -255,32 +272,82 @@ def add_link(self, row):
         doc = Document.get_by_any_pgpid(int(row["object_id"]))
         if not doc:
             self.stats["document_not_found"] += 1
+            self.not_found_documents += [row]
             if self.verbosity > self.v_normal:
                 self.stdout.write(
                     "Document %s not found in database" % row["object_id"]
                 )
             return
 
         # target url: combine base url for this link type with link target from csv
-        target_url = "".join([self.base_url[link_type], row["link_target"]])
+        target_url = "".join([self.base_url[link_type], quote(row["link_target"])])
         # get doc relation for this link type
         doc_relation = self.document_relation[link_type]
 
-        # get source based on the link type
+        # get source and location based on the link type
```

Comment:
all of these use the base filename without the extension; let's generate it once outside the if statements

you'll need to add another `os.path` import for `splitext`

```suggestion
        # get source and location based on the link type
        
        basename = splitext(row["link_target"])[0]
```

Reply:
Quick question on this one, should it be that, or `splitext(basename(row["link_target"]))[0]`? I think it depends on whether we want:
```
"6E.1.5 Portrait of India trade pt 2/T-S 10J16.1_4 (PGPID 7892)"
"6E.1.5 Portrait of India trade pt 2/T-S 10J16.1_8 (PGPID 7892)"
"6E.1.5 Portrait of India trade pt 2/T-S 10J16.1_6 (PGPID 7892)"
```
or
```
"T-S 10J16.1_4 (PGPID 7892)"
"T-S 10J16.1_8 (PGPID 7892)"
"T-S 10J16.1_6 (PGPID 7892)"
```

Reply:
ooh, right ! — that's the one case where you shouldn't use the basename

Reply:
thanks for catching — probably worth a comment to note why this one is different than the others

Reply:
Just to clarify, using the `basename` function would give the second one, and not using it would give the first… which one do we want? If we want the first one, then the suggestion you wrote (without `basename`) is fine. I guess a better question is, are there examples where we _do_ want to use the `basename` function? It seems many Goitein notes are like this—should we just apply it to all except the Goitein notes?

Reply:
sorry for being ambiguous — we want the full location, and in fact the first part is more informative than the second

however, I looked at the links csv again and I don't think we need to use `basename` at all — right? these are the only ones that have a nested path, and we want to keep it

Reply:
Great! Then your original suggestion is correct as is, I'll go ahead and use it 😄

Reply:
😆 😂 ok! (the variable name may be slightly confusing!)